inc/qrcode.class.php

```diff
@@ -85,7 +85,20 @@ function generateQRcode($itemtype, $items_id, $rand, $number, $data) {
          $a_content[] = 'Name = '.$item->fields['name'];
       }
       if ($data['url']) {
-         $a_content[] = 'URL = '.$CFG_GLPI["url_base"].Toolbox::getItemTypeFormURL($itemtype, false)."?id=".$items_id;
+         $normal=true;
+         $plugin = new Plugin();
+         if ($plugin->isActivated("genericobject")) {
+            foreach (PluginGenericobjectType::getTypes() as $tmp => $value) {
+               if ($value['itemtype']==$itemtype) {
+                  $a_content[] = 'URL = '.$CFG_GLPI["url_base"]."/plugins/genericobject/front/object.form.php?itemtype=".$itemtype."&id=".$items_id;
+                  $normal=false;
+                  break;
+               }
+            }
+         }
+         if ($normal) {
+            $a_content[] = 'URL = '.$CFG_GLPI["url_base"].Toolbox::getItemTypeFormURL($itemtype, false)."?id=".$items_id;
```

Comment:
Maybe using `$itemtype::getFormURLWithID($items_id)` would prevent adding all this logic. I think generic object are correctly generating their own URL when using this function.

Reply:
I have tried with the recommended function and it works correctly with generic object